### ISSUE TITLE
Suggestion : Add a compiler flag to treat errors as warnings

### ISSUE THREAD

**Issue by ifle** [microsoft/TypeScript#14501]
Add a compiler flag to treat errors as warnings.
That will be very helpful in case of migration from legacy javascript project to typescript

**Comment by RyanCavanaugh** [comment]
Other than printing
```ts
a.ts(12,12): warning TS2345
```
instead of
```ts
a.ts(12,12): error TS2345
```
what would you expect this flag to do, behaviorally?

**Comment by ifle** [comment]
That will allows to compile asp.net web project in VS without compile errors. 
For now the build is fail. We have the legacy asp.net project . We started migration from js to ts. In the first stage we renamed all js files to ts. There are a lot typescript errors, but generated js is still valid.

**Comment by mribichich** [comment]
Same for me, I have a couple of errors that I know there ok, but since typescript treats them as errors the build fails.

**Comment by mhegazy** [comment]
what are the errors if i may ask?

**Comment by onzag** [comment]
How is this going, I have thousand of errors because I am running a migration, but the build fails because of the errors, giving a lot of log messages in webpack, I simply can't fit them all.

**Comment by mhegazy** [comment]
> How is this going, I have thousand of errors because I am running a migration, but the build fails because of the errors, giving a lot of log messages in webpack, I simply can't fix them all.

what is your build driver? MSBuild?

**Comment by aminpaks** [comment]
I think the new compiler `@ts-ignore` flag will cover this no?

**Comment by ifle** [comment]
That is right direction :). In my case the best solution is adding the flag to the tsconfig with optional list of ignored errors

**Comment by aminpaks** [comment]
@ifle apparently the topic is more complicated that I thought. [Read](https://gist.github.com/alexeagle/349887ddb277aa5833643fc8c0b5465b) what @alexeagle has to say about this

**Comment by alexeagle** [comment]
I agree, for violations of the type system or language spec, `// @ts-ignore` gives you an extra freedom to disable errors during a migration.

**Comment by KodaPOwER** [comment]
Any news on this?
How can i just compile even on error? We are migrating from js to ts. 
I have a lot of file just renamed to .ts waiting to be converted.
If i compile on save, i have the error in the error list but the js file is emitted. 
Instead msbuild stop on the error and that broke my CI in teamcity. 
Why i can't replicate this behavior in msbuild? I can't flag 2000+ lines with // @ts-ignore

**Comment by bodinsamuel** [comment]
Hello, It's a been long time and the issue is not closed.
Is it something you are planning? thanks :)

**Comment by RyanCavanaugh** [comment]
I think the only thing left to address here is the msbuild scenario, which is effectively tracked at #13280

**Comment by QingLeiLi** [comment]
Maybe we add a ts config option to prevent the error to throw.

**Comment by nbkhope** [comment]
Was this feature ever implemented? I too would love to have errors be just warnings as it helps migrating a codebase from JS to TS.

**Comment by TomasHubelbauer** [comment]
Would be cool to be able to treat unused imports as errors.
